documents/building/dev_celix_with_clion.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Building and Developing Apache Celix with CLion
+---
+
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+   
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Building and Developing Apache Celix with CLion
+Apache Celix can be build for development in CLion with use of the Conan package manager.
+Conan will arrange the building of the Apache Celix dependencies and generate Find<package> files for these dependencies.
+
+Conan will also generate a `active_run.sh` and `deactivate_run.sh` script that does the environment (de)setup for the 
+binary locations of the build dependencies (i.e. configures `PATH` and `LD_LIBRARY_PATH`/`DYLD_LIBRARY_PATH`).
+
+## Setting up the build directory
+```shell
+#clone git repo
+git clone https://github.com/apache/celix.git
+cd celix
+
+#if needed setup conan default and debug profile
+conan profile new default --detect
+conan profile new debug --detect
+conan profile update settings.build_type=Debug debug
+
+#generate and configure cmake-build-debug directory
+conan install . celix/2.3.0@apache/testing -pr:b default -pr:h debug -if cmake-build-debug/ -o celix:enable_testing=True -o celix:enable_address_sanitizer=True -o celix:build_all=True -b missing
```

Comment:
For the conan ref is used `celix/2.3.0@apache/testing`, because I was not sure if `celix/2.3.0@zhengpeng/testing` is correct ref to use. But I am happy to revert back to `celix/2.3.0@zhengpeng/testing`

Reply:
I use zhengpeng/testing in private conan-server to avoid conflict with upstream official Celix conan package to be released. IMO, user/channel is unnecessary for an upstream package.